src/System.Private.ServiceModel/tests/Common/Infrastructure/Framework.cs

```diff
@@ -0,0 +1,94 @@
+        private const string NetCoreFrameworkName = ".NET Core";
+
+        private static FrameworkID _currentFrameworkID = 0;
+        private static string _currentFrameworkDescription;
+
+        private static string CurrentFrameworkDescription
+        {
+            get
+            {
+                if (_currentFrameworkDescription == null)
+                {
+                    _currentFrameworkDescription = RuntimeInformation.FrameworkDescription;
+                }
+
+                return _currentFrameworkDescription;
+            }
+        }
+
+        public static FrameworkID Current
+        {
+            get
+            {
+                if (_currentFrameworkID == 0)
+                {
+                    _currentFrameworkID = DetectCurrentFramework();
+                    Console.WriteLine(String.Format("Detected current framework as \"{0}\" from description \"{1}\"",
+                                     _currentFrameworkID.Name(), CurrentFrameworkDescription));
+                }
+
+                return _currentFrameworkID;
+            }
+        }
+
+        // Extension method that returns 'true' if the given FrameworkID
+        // includes the FrameworkID on which this code is currently executing.
+        public static bool MatchesCurrent(this FrameworkID frameworkID)
+        {
+            return ((Current & frameworkID) != 0);
+        }
+
+        // Extension method for FrameworkID to provide name.
+        // Format will expand to enum's name or collection if multiple.
+        public static string Name(this FrameworkID frameworkID)
+        {
+            return Enum.Format(typeof(FrameworkID), frameworkID, "G");
+        }
+
+        // Detects the FrameworkID associated with the framework
+        // on which this code is running.
+        private static FrameworkID DetectCurrentFramework()
+        {
+            string frameworkName = CurrentFrameworkDescription;
+            if (frameworkName.IndexOf(NetNativeFrameworkName, StringComparison.Ordinal) >= 0)
+            {
+                return FrameworkID.NetNative;
+            }
+
+            if (frameworkName.IndexOf(NetFrameworkFrameworkName, StringComparison.Ordinal) >= 0)
+            {
+                return FrameworkID.NetFramework;
+            }
+
+            if (frameworkName.IndexOf(NetCoreFrameworkName, StringComparison.Ordinal) >= 0)
+            {
+                return FrameworkID.NetCore;
+            }
+
+            // We adopt the ID "unknown" whenever we cannot detect the framework to avoid
+            // throwing exceptions during test discovery when new frameworks become available.
+            // The "unknown" in the skip reasons will indicate investigation is required.
+            Console.WriteLine(String.Format(
+                                "FrameworkID cannot be detected from RuntimeInformation.FrameworkDescription \"{0}\".", 
+                                frameworkName));
```

Comment:
Same comment as above -- leaving in at least for now because it provides vital information in the console logs across all the platforms.